src/library/scala/collection/mutable/BitSet.scala

```diff
@@ -171,6 +182,113 @@ class BitSet(protected[collection] final var elems: Array[Long])
   override def zip[B](that: IterableOnce[B])(implicit @implicitNotFound(collection.BitSet.zipOrdMsg) ev: Ordering[(Int, B)]): SortedSet[(Int, B)] =
     super.zip(that)
 
+  override def addAll(xs: IterableOnce[Int]): this.type = xs match {
+    case bs: collection.BitSet => this |= bs
+    case range: Range =>
+      if (range.nonEmpty) {
+        val start = range.min
+        if (start >= 0) {
+          val end = range.max
+          val endIdx = end >> LogWL
+          ensureCapacity(endIdx)
+
+          if (range.step == 1 || range.step == -1) {
+            val startIdx = start >> LogWL
+            val wordStart = startIdx * BitSetOps.WordLength
+            val wordMask = -1L << (start - wordStart)
+
+            if (endIdx > startIdx) {
+              elems(startIdx) |= wordMask
+              java.util.Arrays.fill(elems, startIdx + 1, endIdx, -1L)
+              elems(endIdx) |= -1L >>> (BitSetOps.WordLength - (end - endIdx * BitSetOps.WordLength) - 1)
+            } else {
+              elems(endIdx) |= (wordMask & (-1L >>> (BitSetOps.WordLength - (end - wordStart) - 1)))
+            }
+            this
+          } else {
+            super.addAll(range)
+          }
+        } else {
+          super.addAll(range)
+        }
+      } else {
+        this
+      }
+
+    case sorted: collection.SortedSet[Int] =>
+      // if `sorted` is using the regular Int ordering, ensure capacity for the largest
+      // element up front to avoid multiple resizing allocations
+      if (sorted.nonEmpty) {
+        val ord = sorted.ordering
+        if (ord eq Ordering.Int) {
+          ensureCapacity(sorted.lastKey >> LogWL)
+        } else if (ord eq Ordering.Int.reverse) {
+          ensureCapacity(sorted.firstKey >> LogWL)
+        }
+        val iter = sorted.iterator
+        while (iter.hasNext) {
+          addOne(iter.next())
+        }
+      }
+
+      this
+
+    case pq: PriorityQueue[Int] =>
```

Comment:
Are we sure this is necessary?  It's certainly possible, but is `PriorityQueue[Int]` really used enough to be worth the effort?